### PR TITLE
Fixes #11716: Synchronize stdlib from masterfiles

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/11716

### CODE REVIEW

File: tree/20_cfe_basics/cfengine/common.cf [hunk 1]
```diff
@@ -302,33 +300,175 @@ body classes classes_generic(x)
...
 {
       promise_repaired => { "promise_repaired_$(x)", "$(x)_repaired", "$(x)_ok", "$(x)_reached" };
-      repair_failed => { "repair_failed_$(x)", "$(x)_failed", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
-      repair_denied => { "repair_denied_$(x)", "$(x)_denied", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
-      repair_timeout => { "repair_timeout_$(x)", "$(x)_timeout", "$(x)_not_ok", "$(x)_error","$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
-      promise_kept => { "promise_kept_$(x)", "$(x)_kept", "$(x)_ok", "$(x)_not_repaired", "$(x)_reached" };
+      repair_failed => { "repair_failed_$(x)", "$(x)_failed", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_reached" };
+      repair_denied => { "repair_denied_$(x)", "$(x)_denied", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_reached" };
+      repair_timeout => { "repair_timeout_$(x)", "$(x)_timeout", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_reached" };
+      promise_kept => { "promise_kept_$(x)", "$(x)_kept", "$(x)_ok", "$(x)_reached" };
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/common.cf]: we are using also _not_repaired in these 4 classes

File: tree/20_cfe_basics/cfengine/common.cf [hunk 2]
```diff
@@ -302,33 +300,175 @@ body classes classes_generic(x)
...
 {
       scope => "$(scope)";
       promise_repaired => { "promise_repaired_$(x)", "$(x)_repaired", "$(x)_ok", "$(x)_reached" };
-      repair_failed => { "repair_failed_$(x)", "$(x)_failed", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
-      repair_denied => { "repair_denied_$(x)", "$(x)_denied", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
-      repair_timeout => { "repair_timeout_$(x)", "$(x)_timeout", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
-      promise_kept => { "promise_kept_$(x)", "$(x)_kept", "$(x)_ok", "$(x)_not_repaired", "$(x)_reached" };
+      repair_failed => { "repair_failed_$(x)", "$(x)_failed", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_reached" };
+      repair_denied => { "repair_denied_$(x)", "$(x)_denied", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_reached" };
+      repair_timeout => { "repair_timeout_$(x)", "$(x)_timeout", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_reached" };
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/common.cf]: same here

File: tree/20_cfe_basics/cfengine/files.cf [hunk 3]
```diff
@@ -361,20 +404,20 @@ bundle edit_line manage_variable_values_ini(tab, sectionName)
 
       # If the line is there, but commented out, first uncomment it
       "#+\s*$(index)\s*=.*"
-      select_region => INI_section("$(sectionName)"),
+      select_region => INI_section(escape("$(sectionName)")),
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/files.cf]: i'm amazed it works ...

**Comment by amousset** [comment]
Commit modified

File: tests/acceptance/30_generic_methods/unsafe/service_ensure_running.cf [hunk 4]
```diff
@@ -22,8 +22,10 @@ body common control
 bundle agent init
 {
   vars:
-    !aix::
+    debian::
```
Comment (ncharles) [inline tests/acceptance/30_generic_methods/unsafe/service_ensure_running.cf]: this should not change

File: tree/10_ncf_internals/modules/packages/apt_get [hunk 5]
```diff
@@ -193,7 +193,7 @@ def list_updates(online):
         #                repository(ies)      arch (might be optional)
         #                       |               |
         #                    /--+-\   /---------+---------\
-                         "\s+[^[)]*(\s\[(?P<arch>[^]\s]+)\])?\).*", line)
+                         "(?:\s+\S+)*?(\s+\[(?P<arch>[^]\s]+)\])?\).*", line)
```
Comment (ncharles) [inline tree/10_ncf_internals/modules/packages/apt_get]: Do you know why it changed ?

**Comment by amousset** [comment]
Commit modified

File: tree/20_cfe_basics/README.md [hunk 6]
```diff
@@ -4,3 +4,10 @@
 
 This directory contains librairies with utility bodies that can be reused. Most notably, it contains the cfengine standard librairy.
 
+The `cfengine` folder contains the CFEngine stdlib from 3.10.
+
+The changes made are:
+
+* Removing some files we do not use: testing.cf, autorun.cf, stdlib.cf, examples.cf, cfe_internal_hub.cf, cfengine_enterprise_hub_ha.cf
```
Comment (ncharles) [inline tree/20_cfe_basics/README.md]: if the goal is to use integration with existing installations, we should keep autorun, cfe_internal_hub and cfengine_enterprise_hub_ha

**Comment by amousset** [comment]
Commit modified

File: tree/10_ncf_internals/modules/packages/apt_get (hunk 5 above)
Comment (amousset) [reply tree/10_ncf_internals/modules/packages/apt_get]: ENT-3438: make apt_get package module work with repositories containing spaces in the label

Comment (amousset) [reply tree/10_ncf_internals/modules/packages/apt_get]: I will backport this to 4.1 as well

File: tree/20_cfe_basics/cfengine/feature.cf [hunk 7]
```diff
@@ -0,0 +1,87 @@
+bundle agent feature
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/feature.cf]: what is the goal of this bundle ?

**Comment by amousset** [comment]
Commit modified

File: tree/20_cfe_basics/cfengine/paths.cf [hunk 8]
```diff
@@ -101,12 +104,14 @@ bundle common paths
     _have_bin_systemctl::
       "path[systemctl]"      string => "/bin/systemctl";
     !_have_bin_systemctl::
-      "path[systemctl]"      string => "/usr/bin/systemctl";
+      "path[systemctl]"      string => "/bin/systemctl";
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/paths.cf]: i'm pretty sure this is not what we want:
` "_have_bin_systemctl" expression => fileexists("/bin/systemctl");`

**Comment by amousset** [comment]
Commit modified

File: tree/20_cfe_basics/cfengine/paths.cf [hunk 9]
```diff
@@ -101,12 +104,14 @@ bundle common paths
...
 
     linux::
       "path[lsattr]"        string => "/usr/bin/lsattr";
       "path[tar]"           string => "/bin/tar";
-      "path[chkconfig]"     string => "/sbin/chkconfig";
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/paths.cf]: can't we have chkconfig on other system than redhat-like, debian-like, and suse?

**Comment by amousset** [comment]
Commit modified

File: tree/20_cfe_basics/cfengine/paths.cf (hunk 9 above)
Comment (amousset) [reply tree/20_cfe_basics/cfengine/paths.cf]: Added in https://github.com/Normation/ncf/commit/78b8e471648d18c83b826ddc6fbf80d08675f388 but it does not seem necessary after the changes in the generic method.

File: tree/20_cfe_basics/cfengine/paths.cf (hunk 8 above)
Comment (amousset) [reply tree/20_cfe_basics/cfengine/paths.cf]: Fixed and opened https://github.com/cfengine/masterfiles/pull/1004

**Comment by ncharles** [comment]
tree/10_ncf_internals/modules/packages/zypper should be 755 to be consistant

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

File: tree/20_cfe_basics/cfengine/bundles.cf [hunk 10]
```diff
@@ -39,14 +39,29 @@
...
+      "inputs" slist => { "$(this.promise_dirname)/paths.cf",
+                          "$(this.promise_dirname)/files.cf",
+                          "$(this.promise_dirname)/commands.cf" };
+}
+
+body file control
+# @ignore
+{
+      inputs => { @(bundles_common.inputs) };
+}
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/bundles.cf]: shouldn't we remove these ? do you know if they conflict with our loading method ?

File: tree/20_cfe_basics/cfengine/cfe_internal.cf [hunk 11]
```diff
@@ -39,100 +39,87 @@
 # the online docs
 #
 
-# For CFEngine Core: 3.5.0 to 3.5.x
-# Internal hub maintenance bundles
-
+# For CFEngine Core: 3.8.0 to 3.8.x
```
Comment (ncharles) [inline tree/20_cfe_basics/cfengine/cfe_internal.cf]: i thought we were advertising compatibility for 3.7 ?

Comment (amousset) [reply tree/20_cfe_basics/cfengine/cfe_internal.cf]: We are, these comments are likely wrong (or it is a bug in the masterfiles). I will check with a 3.7 agent.

**Pull request merged**